components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -815,4 +816,29 @@ private String getIdpSessionKey(String accessToken) {
         }
         return idpSessionKey;
     }
+
+    /**
+     * Check the requirement of having a configured public certificate or JWKS in SP and throw an exception with an
+     * error message if the public certificate or JWKS in SP is not configured.
+     *
+     * @param clientId Client ID of the service provider.
+     * @param tenantDomain Tenant domain of the service provider.
+     * @throws IdentityOAuth2Exception
+     */
+    private void checkIfPublicCertConfigured(String clientId, String tenantDomain) throws IdentityOAuth2Exception {
```

Comment:
Better to change the name since the name implies only public certificate. Ideally the name should consider both cases JWKS url and public ceritificate

Reply:
In UI, we refer to both the JWKS endpoint and public certificate as `Certificate`.<img width="530" alt="Screenshot 2022-04-11 at 3 27 31 PM" src="https://user-images.githubusercontent.com/47919820/162734685-35b6937c-bd62-40dc-8707-9f0ada73b6ff.png">